platforms/jvm/plugins-java/src/integTest/groovy/org/gradle/java/JavaCrossCompilationIntegrationTest.groovy

```diff
@@ -173,69 +87,47 @@ class JavaCrossCompilationIntegrationTest extends AbstractIntegrationSpec {
             }
         """
 
-        expect:
-        succeeds 'run'
-        output.contains("java home: ${FileUtils.canonicalize(target.javaHome)}")
-        output.contains("java version: ${version}")
-
-        where:
-        version << allJavaVersions()
-    }
-
-    def "can generate Javadocs using target Java version"() {
-        given:
-        withJavaProjectUsingToolchainsForJavaVersion(version)
+        when:
+        withInstallations(jdk)
+        succeeds("run")
 
-        expect:
-        succeeds 'javadoc'
-        file('build/docs/javadoc/Thing.html').text.matches("(?s).*Generated by javadoc \\(.*?\\Q${version}\\E.*")
+        then:
+        output.contains("java home: ${FileUtils.canonicalize(jdk.javaHome)}")
+        output.contains("java version: ${jdk.javaVersion}")
 
         where:
-        version << allJavaVersions()
+        jdk << AvailableJavaHomes.allJdkVersions
     }
 
-    def "can compile but cannot run tests against Java #version JDK"() {
```

Comment:
negative test coverage across java versions is verified by `TestTaskJavaVersionIntegrationTest`